packages/peregrine/lib/talons/CartPage/ProductListing/useProductListing.js

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../context/cart';
+import { useEffect } from 'react';
+
+export const useProductListing = props => {
+    const { query } = props;
+
+    const [{ cartId }] = useCartContext();
+
+    const { data, error, loading } = useQuery(query, {
+        variables: { cartId }
+    });
+
+    let items = [];
+    if (!error && !loading) {
+        items = data.cart.items;
+    }
+
+    useEffect(() => {
```

Comment:
This is reachable code, but I like to put effects/hooks at the top since they cannot be conditional. Not super important.

Reply:
Moved in f622664